File: neuracx-obd-sdk/nodejs/examples/basic-usage.js
'use strict';

// Set these in your environment before running this example:
//   OBD_API_BASE_URL, OBD_CLIENT_ID, OBD_API_KEY
const { OBDClient, OBDApiError } = require('omni-obd-sdk');

const client = new OBDClient({
  baseUrl: process.env.OBD_API_BASE_URL,
  clientId: process.env.OBD_CLIENT_ID,
  apiKey: process.env.OBD_API_KEY,
});

async function main() {
  try {
    const response = await client.initiateCall({
      fromNumber: '+9104847189769',
      toNumber: '+916238330634',
      dialRequestExpiry: 10,
      refId: 6565,
      timeout: 30,
      nextActionByApi: false,
      nextActionUrl: 'http://your-ivr-server.example.com/ivr/webhook',
      stream: {
        enabled: true,
        record: true,
        streamUrl: 'ws://your-media-server.example.com:3031',
        duration: 10,
        chunkSize: 1600,
        startPhase: 'ringing',
        customParam: {
          customerId: 'CUST10001',
          orderId: 'ORD89231',
        },
      },
      pingbackUrl: 'http://your-server.example.com/api/postdata',
      callEvents: {
        initiated: true,
        ringing: true,
        answered: true,
        completed: true,
        failed: true,
        expired: true,
      },
    });

    console.log('Call initiated:', response);
  } catch (err) {
    if (err instanceof OBDApiError) {
      console.error(`OBD API error (status ${err.statusCode}):`, err.message, err.responseBody);
    } else {
      throw err;
    }
  }
}

main();

File: neuracx-obd-sdk/python/examples/basic_usage.py
import os

from obd_sdk import OBDApiError, OBDClient, build_call_events, build_stream_config

# Set these in your environment before running this example:
#   OBD_API_BASE_URL, OBD_CLIENT_ID, OBD_API_KEY
client = OBDClient(
    base_url=os.environ["OBD_API_BASE_URL"],
    client_id=os.environ["OBD_CLIENT_ID"],
    api_key=os.environ["OBD_API_KEY"],
)

try:
    response = client.initiate_call(
        from_number="+9104847189769",
        to_number="+916238330634",
        dial_request_expiry=10,
        ref_id=6565,
        timeout=30,
        next_action_by_api=False,
        next_action_url="http://your-ivr-server.example.com/ivr/webhook",
        stream=build_stream_config(
            stream_url="ws://your-media-server.example.com:3031",
            record=True,
            duration=10,
            chunk_size=1600,
            custom_param={"customer_id": "CUST10001", "order_id": "ORD89231"},
        ),
        pingback_url="http://your-server.example.com/api/postdata",
        call_events=build_call_events(),
    )
    print("Call initiated:", response)
except OBDApiError as exc:
    print(f"OBD API error (status {exc.status_code}): {exc}")
    print(exc.response_body)
